website/content/docs/integrations/vault-integration.mdx

```diff
@@ -24,21 +22,148 @@ install Vault separately from Nomad. Nomad does not run Vault for you.
 
 -> **Note:** Vault integration requires Vault version 0.6.2 or higher.
 
-## Vault Configuration
+## Agent Configuration
+
+To enable Vault integration, please see the [Nomad agent Vault
+integration][config] configuration.
+
+## Vault Definition Syntax
+
+To configure a job to retrieve Vault tokens, please see the [`vault` job
+specification documentation][vault-spec].
+
+## Using Nomad Workload Identity with Vault
+
+Starting in Nomad 1.7, Nomad clients will use a task's [Workload Identity][] to
+authenticate to Vault and obtain a Vault token specific to the task.
+
+### Configuring Vault Authentication
+
+To configure Vault for use with Nomad workload identity, you'll enable the Vault
+JWT backend, create a Vault role and auth method, and finally create Vault
+policies to that map Nomad workload claims to secrets in Vault.
+
+<EnterpriseAlert>Nomad Enterprise users who are using Vault Namespaces should
+note that you will need to pass the -namespace flag for each of the commands
+below. Each Vault namespace needs its own auth method, role, and
+policies.</EnterpriseAlert>
+
+First enable the JWT authentication backend.
+
+```shell-session
+$ vault auth enable jwt
+```
+
+Create a configuration file for a default Vault [Role][]. This configuration
+maps each Nomad job to its own Vault user claim. You can set the `token_ttl` and
+`token_policies` to whatever value is suitable for your environment. A
+recommended policy is shown later.
+
+```json
+{
+  "role_type": "jwt",
+  "bound_audiences": "vault.io",
+  "user_claim": "/nomad_job_id",
+  "user_claim_json_pointer": true,
+  "claim_mappings": {
+    "nomad_namespace": "nomad_namespace",
+    "nomad_job_id": "nomad_job_id"
+  },
+  "token_ttl": "30m",
+  "token_type": "service",
+  "token_policies": ["nomad-workloads"]
+}
+```
+
+Using that configuration file, create the Vault role.
+
+```shell-session
+$ vault write auth/jwt/role/nomad-workloads @vault-role.json
+```
+
+Create a configuration file for a Vault [JWT Auth Method][]. The `jwks_url`
+field should point to the Nomad servers and the `default_role` should match the
+name of the role you created above.
+
+```json
+{
+  "jwks_url": "https://nomad.example.com:4646/.well-known/jwks.json",
```

Comment:
Should we add some sort of a "suggestion" or "tip" here about having a proxy for this, or some way of assuring HA in case a particular Nomad server goes down? Not really sure this is the right place for it, just thinking out loud.

Reply:
Yeah that's a good idea. Will add that here and to the Consul docs if it's not there already.